src/main/java/com/timgroup/statsd/StatsDNonBlockingProcessor.java

```diff
@@ -0,0 +1,121 @@
+package com.timgroup.statsd;
+
+import java.io.IOException;
+import java.net.SocketAddress;
+import java.nio.ByteBuffer;
+import java.nio.channels.DatagramChannel;
+import java.nio.charset.Charset;
+import java.util.concurrent.Callable;
+
+import java.util.Queue;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class StatsDNonBlockingProcessor extends StatsDProcessor {
+
+    private final Queue<String> messages;
+    private final int qCapacity;
+    private final AtomicInteger qSize;  // qSize will not reflect actual size, but a close estimate.
+
+    StatsDNonBlockingProcessor(final int queueSize, final StatsDClientErrorHandler handler,
+            final int maxPacketSizeBytes, final int poolSize, final int workers)
+            throws Exception {
+
+        super(queueSize, handler, maxPacketSizeBytes, poolSize, workers);
+        this.qSize = new AtomicInteger(0);
+        this.qCapacity = queueSize;
+        this.messages = new ConcurrentLinkedQueue<String>();
+    }
+
+    @Override
+    boolean send(final String message) {
+        if (!shutdown) {
+            if (qSize.get() < qCapacity) {
+                messages.offer(message);
+                qSize.incrementAndGet();
+                return true;
+            }
+        }
```

Comment:
Would it make sense to catch `InterruptedException` as in https://github.com/DataDog/java-dogstatsd-client/pull/96/files#diff-d87f261783303f309bda2086a2e14756R35?

Reply:
I wouldn't expect an `InterruptedException` to happen there as we are dealing with a non-blocking queue in this case and the `offer()` method would just return as opposed to stay there waiting for the write.